src/io/io_routines.f90

```diff
@@ -816,6 +828,8 @@ subroutine io_write3d(filename,varname,data_out, dimnames)
 
         ! Create the variable returns varid of the data variable
         call check( nf90_def_var(ncid, varname, NF90_REAL, dimids, varid), trim(filename)//":"//trim(varname))
+        call check( nf90_def_var_deflate(ncid, varid, shuffle = 1, deflate = 1, deflate_level = 3), trim(filename)//":"//trim(varname))
```

Comment:
make 1, 1, and 3 into at least module level parameters so you don't have to change them everywhere if you want to change them anywhere.